src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -92,10 +95,18 @@ sync_primitives::ConditionalVariable state_condition_;
 }  // namespace
 
 class ApplicationManagerImplTest : public ::testing::Test {
+ public:
+    ApplicationManagerImplTest():
+      mock_message_helper_(application_manager::MockMessageHelper::message_helper_mock())
+      //message_helper_(new MessageHelper::MessageHelper())
+    {}
+
  protected:
   void SetUp() OVERRIDE {
     app_id_ = 10;
 
+    Mock::VerifyAndClearExpectations(&mock_message_helper_);
```

Comment:
@pold500 About this command:  "Mock::VerifyAndClearExpectations(&mock_message_helper_);" it is better to place it in constructor AND in destructor in classes which use mock_message_helper_, and so it is better to move it out from SetUp and TearDown methods.